Rover: added is_omni_rover function

Ammarf opened the pull request:
- Added is_omni_rover function
- Pre_arm_check method checks for 1 or 2 assigned functions

rmackay9 commented:
This looks good to me so I'm happy for it to go into master (assuming that it passes Travis and Semaphore).  There are a few minor style like things that I would have done differently but these aren't blockers for it going in:

- in the pre-arm checks, instead of three checks I probably would have used a counter (incremented each time a motorX is enabled) and then done the report if the counter was 1 or 2.  Still, this is not a functional change, and it really doesn't matter.
- I would have formatted, "if(!(is_omni_rover())) {" ad "if (!is_omni_rover()) {".  i.e. an extra space after the "if" and one less set of brackets.
- instead of declaring "int16_t motor_1" at the top of output_omni and then calculating the value later, I would have done a "const float motor_1 = blahblahblah;".
- there's an extra blank line just after, "SRV_Channels::set_output_pwm(SRV_Channel::k_motor3, motor_3);"

.. but like I said, these are all minor formatting issues and shouldn't hold it up going in.  Let's give it another day or so to see if any other devs like @khancyr or @peterbarker want to provide more feedback.

rmackay9 commented:
Actually a couple of extra items have come up that @Ammarf is going to look at.  I write them here so other viewers don't need to point them out as well:

- in output_omni, steering and throttle are passed in but then immediately overwritten with calculations based on \_throttle and \_steering
- instead of using set_output_pwm, it would be better to extend output_throttle to handle k_motor1, k_motor2, k_motor3

Ammarf commented:
I made the following changes:
- I fixed the issue of the throttle and steering being immediately overwritten in output_omni.
- output_omni now uses output_throttle to set the motors output instead of set_output_pwm.
- Added k_motor1, k_motor2 and k_motor3 to setup_servo_output.
- Fixed some formatting issues that Randy pointed out previously.

rmackay9 commented:
yes, this looks good to me.

I'll giver it some more time in case @khancyr or @OXINARF wants to review it.

khancyr commented on APMrover2/AP_MotorsUGV.cpp:
```diff
@@ -396,6 +423,39 @@ void AP_MotorsUGV::output_regular(bool armed, float steering, float throttle)
     SRV_Channels::set_output_scaled(SRV_Channel::k_steering, steering);
 }
 
+// output for omni style frames
```
could we add here the equation/math where the mixing come from ? or a small explanation on how it works ?

khancyr commented:
Humm, still some little space to add between the math operator ... I will push a patch.
Otherwise for understanding, I think it will be good to have the equation of the mixing in comment otherwise it is hard to understand how it is done.
Maybe we should also get the real range and trim value instead of the hardcore standard value ?

Ammarf commented:
I added the comments to output_omni. I think it's better to add the real range and trim values changes once support for lateral movement is included.

rmackay9 commented:
Merged, thanks!
